libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -461,35 +477,54 @@ bool NavEKF3_core::readyToUseBodyOdm(void) const
 // return true if the filter to be ready to use gps
 bool NavEKF3_core::readyToUseGPS(void) const
 {
-    return validOrigin && tiltAlignComplete && yawAlignComplete && (delAngBiasLearned || assume_zero_sideslip()) && gpsGoodToAlign && (frontend->_fusionModeGPS != 3) && gpsDataToFuse && !gpsInhibit;
+    if (frontend->sources.getPosXYSource() != AP_NavEKF_Source::SourceXY::GPS) {
```

Comment:
No, I think they are consistent.  the confusion may come from the "useVelXYSource()" vs "getPosXYSource()".  For velocities it is possible (at Paul's request) that all configured velocities are fused.  For position only a single source will ever be fused.  The first couple of lines of readyToUseBodyOdm() check if the user has configured the sources so that at least one of EXTNAV or WHEEL_ENCODER may be fused.

In any case, I've added this comment, "// exit immediately if sources not configured to fuse external nav or wheel encoders"